source/base/data_out_base.cc

```diff
@@ -720,143 +707,117 @@ namespace
                      const unsigned int z_offset);
 
     /**
-     * Do whatever is necessary to
-     * terminate the list of cells.
+     * Do whatever is necessary to terminate the list of cells. This function is
+     * usually only overloaded if deal.II is compiled with zlib.
      */
-    void flush_cells ();
+    void flush_cells () {}
 
     /**
-     * Write a complete set of
-     * data for a single node.
+     * Write a complete set of data for a single node.
      *
-     * The index given as first
-     * argument indicates the
-     * number of a data set, as
-     * some output formats require
-     * this number to be printed.
+     * The index given as first argument indicates the number of a data set, as
+     * some output formats require this number to be printed.
      */
     template<typename data>
-    void write_dataset (const unsigned int       index,
-                        const std::vector<data> &values);
+    void write_dataset (const unsigned int index, const std::vector<data> &values);
 
     /**
-     * Forwarding of output stream
+     * Forwarding of an output stream. This function is usually only overloaded
+     * if inheriting classes use zlib.
      */
     template <typename T>
-    std::ostream &operator<< (const T &);
+    std::ostream &operator<< (const T &t)
+    {
+      stream << t;
+      return stream;
+    }
 
-  private:
     /**
-     * The ostream to use. Since
-     * the life span of these
-     * objects is small, we use a
-     * very simple storage
-     * technique.
+     * Since the GMV and Tecplot formats read the x, y and z coordinates in
+     * separate fields, we enable write() to output only a single selected
+     * component at once and do this dim times for the whole set of nodes. This
+     * integer can be used to select the component written.
+     */
+    unsigned int selected_component;
+
+  protected:
+    /**
+     * The ostream to use. Since the life span of these objects is small, we use
+     * a very simple storage technique.
      */
     std::ostream &stream;
 
     /**
-     * The flags controlling the output
+     * The flags controlling the output.
      */
-    const DataOutBase::DXFlags flags;
+    const FlagsType flags;
   };
 
   /**
    * Class for writing basic
-   * entities in @ref SoftwareGMV
-   * format, depending on the
-   * flags.
+   * entities in @ref
+   * SoftwareOpenDX format,
+   * depending on the flags.
    */
-  class GmvStream
+  class DXStream : public StreamBase<DataOutBase::DXFlags>
   {
   public:
-    /**
-     * Constructor, storing
-     * persistent values for
-     * later use.
-     */
-    GmvStream (std::ostream &stream,
-               const DataOutBase::GmvFlags flags);
+    DXStream (std::ostream &stream, const DataOutBase::DXFlags flags);
```

Comment:
I know that this is pre-existing, but can you make the second argument as reference?

Also, we pretty uniformly keep every argument on its own line.